core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -439,12 +439,16 @@ Kokkos::Cuda::initialize WARNING: Cuda is allocating into UVMSpace by default
 //----------------------------------------------------------------------------
 
 using ScratchGrain = Cuda::size_type[Impl::CudaTraits::WarpSize];
-enum { sizeScratchGrain = sizeof(ScratchGrain) };
+static constexpr auto sizeScratchGrain = sizeof(ScratchGrain);
+
+constexpr std::size_t scratch_count(const std::size_t size) {
```

Comment:
The `constexpr` specifier is superfluous.  This will never be evaluated at compile-time.
I think making it a static member function is more pertinent.

Reply:
Is it really worth it to move it into the header since it's only used in the .cpp? What about dropping the `constexpr` and moving it to the anonymous namespace at the top of the file (where we already have helpers like `cuda_kernel_arch`)?

Reply:
I misread and thought it was defined in the body of the class.  But since you are in the anonymous namespace the static specifier is not needed.